src/java.base/share/man/java.1

```diff
@@ -5133,74 +5133,84 @@ When this happens, either the OS or JVM reverts to using regular pages.
 Linux and Windows support large pages.
 .SS Large Pages Support for Linux
 .PP
-The 2.6 kernel supports large pages.
-Some vendors have backported the code to their 2.4\-based releases.
-To check if your system can support large page memory, try the
-following:
+Linux supports large pages since version 2.6.
+To check if your environment supports large pages, try the following:
 .IP
 .nf
 \f[CB]
 #\ cat\ /proc/meminfo\ |\ grep\ Huge
 HugePages_Total:\ 0
 HugePages_Free:\ 0
+\&...
 Hugepagesize:\ 2048\ kB
 \f[R]
 .fi
 .PP
-If the output shows the three "Huge" variables, then your system can
-support large page memory but it needs to be configured.
-If the command prints nothing, then your system doesn\[aq]t support
-large pages.
-To configure the system to use large page memory, login as
-\f[CB]root\f[R], and then follow these steps:
-.IP "1." 3
-If you\[aq]re using the option \f[CB]\-XX:+UseSHM\f[R] (instead of
-\f[CB]\-XX:+UseHugeTLBFS\f[R]), then increase the \f[CB]SHMMAX\f[R] value.
-It must be larger than the Java heap size.
-On a system with 4 GB of physical RAM (or less), the following makes all
-the memory sharable:
-.RS 4
+If the output contains items prefixed with "Huge", then your system
+supports large pages.
+The values may vary depending on environment.
+The \f[CB]Hugepagesize\f[R] field shows the default large page size in
+your environment, and the other fields show details for large pages of
+this size.
+Newer kernels have support for multiple large page sizes.
+To list the supported page sizes, run this:
+.IP
+.nf
+\f[CB]
+#\ ls\ /sys/kernel/mm/hugepages/
+hugepages\-1048576kB\ \ hugepages\-2048kB
+\f[R]
+.fi
+.PP
+The above environment supports 2 MB and 1 GB large pages, but they need
+to be configured so that the JVM can use them.
+When using explicit large pages (options \f[CB]\-XX:+UseSHM\f[R] or
+\f[CB]\-XX:+UseHugeTLBFS\f[R]), the number of large pages must be
+pre\-allocated.
+For example, to enable 8 GB of memory to be backed by 2 MB large pages,
+login as \f[CB]root\f[R] and run:
 .RS
 .PP
-\f[CB]#\ echo\ 4294967295\ >\ /proc/sys/kernel/shmmax\f[R]
+\f[CB]#\ echo\ 4096\ >\ /sys/kernel/mm/hugepages/hugepages\-2048kB/nr_hugepages\f[R]
```

Comment:
- vm.nr_hugepages is not the only way to do this setup, since we have vm.nr_overcommit_hugepages, which are a bit more flexible to use. Is the java man page supposed to be complete in explaining this, or should it only show one possible way of many? 
- most users probably just use UseLargePages and are not aware of the different flavors. So they would not know what "explicit" means. How about, instead of mentioning UseSHM and UseHugeTLBFS, not just: "if you use large pages but don't use transparent huge pages (UseTransparentHugePages), large pages need to be preallocated...". 
- Otherwise, at least reverse the order of the two options? UseSHM is mentioned first, which is maybe historical, but TLBFS is the standard.

Reply:
>     * vm.nr_hugepages is not the only way to do this setup, since we have vm.nr_overcommit_hugepages, which are a bit more flexible to use. Is the java man page supposed to be complete in explaining this, or should it only show one possible way of many?
> 
I think mentioning one way to configure it is fine, we don't want to try to cover every possibility. I mean there are multiple ways of specifying each of the options as well. Most advanced users will likely refer to the kernel documentation for more details. I see this as a short explanation of one way to enable large pages.

>     * most users probably just use UseLargePages and are not aware of the different flavors. So they would not know what "explicit" means. How about, instead of mentioning UseSHM and UseHugeTLBFS, not just: "if you use large pages but don't use transparent huge pages (UseTransparentHugePages), large pages need to be preallocated...".
> 
Something like: 
```
- When using explicit large pages (options `-XX:+UseSHM` or `-XX:+UseHugeTLBFS`), the number of...
+ When using large pages and not enabling transparent huge pages (option `-XX:+UseTransparentHugePages`), the number of...
```
>     * Otherwise, at least reverse the order of the two options? UseSHM is mentioned first, which is maybe historical, but TLBFS is the standard.
Good point, I reversed the order of configuring the tow but forgot it here. But going with the above is better I thing.